### PR TITLE
`azurerm_kubernetes_cluster` - `cpu_cfs_quota_enabled` defaults to `true` in 4.0

### PR DESCRIPTION
**Pull request by ms-henglu**
fixes https://github.com/hashicorp/terraform-provider-azurerm/issues/22424
```
=== RUN   TestAccKubernetesClusterNodePool_kubeletAndLinuxOSConfigPartial
=== PAUSE TestAccKubernetesClusterNodePool_kubeletAndLinuxOSConfigPartial
=== CONT  TestAccKubernetesClusterNodePool_kubeletAndLinuxOSConfigPartial
--- PASS: TestAccKubernetesClusterNodePool_kubeletAndLinuxOSConfigPartial (964.68s)
PASS
```

### CODE REVIEW

File: internal/services/containers/kubernetes_nodepool.go [hunk 1]
```diff
@@ -365,10 +366,16 @@ func schemaNodePoolKubeletConfig() *pluginsdk.Schema {
 			},
 		},
 	}
+
+	if !features.FourPointOhBeta() {
+		schema.Elem.(*pluginsdk.Resource).Schema["cpu_cfs_quota_enabled"].Default = false
+	}
```
Comment (magodo) [inline internal/services/containers/kubernetes_nodepool.go:372]: Can we revert the logic here to set the `Default` to `true` when it is 4.0, and adding a commnet above the `if` as a TODO, reminding the maintainer to add the default value to the document when updating to 4.0?

(same as below)

**Review by magodo — CHANGES_REQUESTED** [review]
Thanks for this PR!
I've taken a look through and left some comments inline, but this is mostly looking good to me 👍

File: internal/services/containers/kubernetes_nodepool.go (hunk 1 above)
Comment (ms-henglu) [reply internal/services/containers/kubernetes_nodepool.go:373]: Got it, I've added comment.

**Review by magodo — APPROVED** [review]

**Review by stephybun — APPROVED** [review]
LGTM 👍

**Pull request merged**